lib/src/main/java/com/hootsuite/nachos/NachoTextView.java

```diff
@@ -727,6 +727,128 @@ public void setTextWithChips(@Nullable List<ChipInfo> chips) {
         endUnwatchedTextChange();
     }
 
+    /**
+     *
+     * @return String of unchipped text that AutoCompleteTextView sees
+     */
+
+    public String getTextForSuggestion() {
+        if (getChipTokenizer() != null) {
+            int end = getSelectionEnd();
+            Editable editable = getText();
+            int start = getChipTokenizer().findTokenStart(editable, end);
+            if (start != -1 && end != -1) {
+                if (end < start) {
+                    start = end;
+                }
+                return editable.subSequence(start, end).toString();
+            }
+        }
+        return "";
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @return true if an existing chip contains text and null object
+     */
+    public boolean isAlreadyAdded(CharSequence text) {
```

Comment:
Agreed. This is a requirement that your application needs, and adding it in here doesn't help describe the `NachoTextView` more. Keep in your forked version or have Util methods in this library to help you with it.